core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFormatterUtil.java

```diff
@@ -201,6 +252,93 @@ static ExprValue getFormattedTime(ExprValue timeExpr, ExprValue formatExpr) {
     return getFormattedString(formatExpr, TIME_HANDLERS, time);
   }
 
+  private static boolean canGetDate(TemporalAccessor ta) {
+    if (!ta.isSupported(ChronoField.YEAR)
+        || !ta.isSupported(ChronoField.MONTH_OF_YEAR)
+        || !ta.isSupported(ChronoField.DAY_OF_MONTH)) {
+      return false;
+    }
+    return true;
+  }
+
+  private static boolean canGetTime(TemporalAccessor ta) {
+    if (!ta.isSupported(ChronoField.HOUR_OF_DAY)
+        || !ta.isSupported(ChronoField.MINUTE_OF_HOUR)
+        || !ta.isSupported(ChronoField.SECOND_OF_MINUTE)) {
+      return false;
+    }
+    return true;
+  }
+
+  private static boolean isValidTemporalAccessor(TemporalAccessor ta) {
+    if (canGetDate(ta) || canGetTime(ta)) {
```

Comment:
You can just do `return canGetDate(ta) || canGetTime(ta);` for these functions

Reply:
Oh yeah haha. Thanks for catching that. This function used to be a lot bigger but slowly reduced in size... I guess now it's just not needed.

Reply:
Fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/231/commits/026c830f001a3b8a463cbb1cf8dab7bd6689d054